libd/src/d/semantic/flow.d

```diff
@@ -213,8 +214,17 @@ public:
 		switchStack = varStack;
 		allowFallthrough = true;
 		switchFunTerminate = true;
+		switchHaveDefault = false;
 		
 		visit(s.statement);
```

Comment:
for the above case before visit(s.statement) switchHaveDefault needs to be set to false